sw/device/lib/dif/dif_aes.c

```diff
@@ -0,0 +1,424 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_aes.h"
+
+#include <stddef.h>
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+
+#include "aes_regs.h"  // Generated.
+
+/*
+ * From: https://docs.opentitan.org/hw/ip/aes/doc/index.html#register-table,
+ * aes.CTRL.
+ */
+
+/*
+ * Field to select AES block cipher mode.
+ */
+typedef enum aes_mode_field_val {
+  kAesModeFieldValEcb = 0x01,  /**< The Electronic Codebook Mode. */
+  kAesModeFieldValCbc = 0x02,  /**< The Cipher Block Chaining Mode. */
+  kAesModeFieldValCfb = 0x04,  /**< TODO */
```

Comment:
What does this TODO mean?

Reply:
It means these modes still need to be implemented. This is tracked in #4833.